File: src/benchmarks/iota_sum_reduce/isr.c
#include <stdio.h>
#include <stdlib.h>

int
compute(int rgt) {
	int i, sum;
	
	sum = 0;
	
	for (i = 0; i < rgt; i++) {
		sum += i;
	}
	
	return sum;
}

int
main(int argc, char *argv[])
{
	int count, i;
	int *res;
	
	count = 100000;
	res = calloc(sizeof(int), 100000);
	
	for (i = 0; i < count; i++) res[i] = compute(i);
	
	printf("%d\n", res[count -1]);
}

File: attic/compiler/v7/benchmarks/iota_sum_reduce/isr.java
public class isr {
	
	private static int compute(int rgt) {
		int sum = 1;
		
		for (int i = 0; i < rgt; i++) sum += i;
		
		return sum;
	}
	
	public static void main(String[] args) {
		int count = 100000;
		int[] res = new int[count];
		
		for (int i = 0; i < count; i++) res[i] = compute(i);
		
		System.out.println(res[count - 1]);
	}
	
}
